pages/api/studentPortal/freemium/showModal.ts

```diff
@@ -0,0 +1,25 @@
+import { differenceInHours } from "date-fns";
+import { useSelector } from "react-redux";
+import { profileSelector } from "../../../../redux/profileSlice";
+
+export function convertDateToUTC(date) { return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate(), date.getUTCHours(), date.getUTCMinutes(), date.getUTCSeconds()); }
+
+export function showModal() {
+    const { userRole, userProfileData } = useSelector(profileSelector);
+    const lastSeenTime = userProfileData.lastSeen;
+    const lastSeenDateTime = new Date(lastSeenTime);
+    const currentDateTime = new Date();
+    const timeDifferenceInHours = differenceInHours(
+      currentDateTime,
+      lastSeenDateTime,
+    );
+
+    if (
+      timeDifferenceInHours <= 24 &&
```

Comment:
I feel like this won't do what you want it to do. Last seen is updated when you load the student portal

Reply:
the goal is to have the modal not come up too often.
I thought if they hadn't loaded the student portal in 24 hours that would  work...and then it would come up.  if they keep logged in and working, they probably won't keep needing the message.

Do you have any ideas or should I create something?

Reply:
Would one solution be to add a column to the users table called lastLogin and in the SignInPage.tsx create a useMutation that writes a time back to the database when the user logs in and then use that value in my showModal function to see if it's been more than 24 hours?  
Another option would be to put the useMutation to write the time back to the database inside the showModal and label it lastSeenModal and then use that time in my showModal function to see if it's been more than 24 hours.

Reply:
I think we should create a new table and have a foreign key that links to the user table

Reply:
and create a useMutation like you said